### PR TITLE
fix/Switch-to-Clang

### PR DESCRIPTION
**Pull request by santiberna**
### Description

This pull request makes the project compile with clang instead of GCC, massivelly speeding up the slow link and compile times we were previously having. It also adds a fix for not seeing std data structures in the debugger.

MISSING: Compilation still does not work because it is necessary to have the dll import libraries (.lib) to be able to link the project with clang

WARNING: To get everything working, follow these instructions:

Make sure you have clang and lldb: I have installed them using Chocolatey - https://community.chocolatey.org/packages/llvm
Use the specified VisualStudio toolchain on CLion:

image

Make sure it is the top one

### Test criteria

Compile and run the project successfully
Use the debugger to inspect and std::vector (for example)

### CODE REVIEW

**Review by MuskettaMan — APPROVED** [review]
Some small comments, but overall looks good

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,21 +68,14 @@ endif ()
...
-        "${PROJECT_SOURCE_DIR}/external/steamworks/lib/$<PLATFORM_ID>"
-        $<TARGET_FILE_DIR:Game>)
-
-# Steam app ID (TODO: Remove from distribution build!)
-
-add_custom_command(TARGET Game POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy
-        "${PROJECT_SOURCE_DIR}/steam_appid.txt"
-		$<TARGET_FILE_DIR:Game>)
+#add_custom_command(TARGET Game POST_BUILD
```
[the remark this answers is not in the document]
Comment (santiberna) [reply CMakeLists.txt:71]: Removed it, the copy to output is handled on the CMakeLists.txt relative to the fmod and steamapi libraries

File: engine/audio/private/audio_module.cpp [hunk 2]
```diff
@@ -1,11 +1,9 @@
 #include "audio_module.hpp"
 
-#include "fmod.h"
-#include "fmod_errors.h"
-#include "fmod_studio.h"
+#include "fmod_debug.hpp"
 #include "log.hpp"
 
-#include "fmod_debug.hpp"
+#include <fmod_include.hpp>
```
[the remark this answers is not in the document]
Comment (santiberna) [reply engine/audio/private/audio_module.cpp:6]: Switched to ""

File: engine/include/ecs.hpp [hunk 3]
```diff
@@ -60,16 +58,13 @@ void save(Archive& archive, ECS const& ecs, MAYBE_UNUSED uint32_t version)
     }
 }
 template <typename T>
-T& ECS::GetSystem()
+T* ECS::GetSystem()
```
[the remark this answers is not in the document]
Comment (santiberna) [reply engine/include/ecs.hpp:61]: Added a comment specifying that if the system is not found then nullptr is returned

**Pull request merged**